### PR TITLE
Add gd32vw55x flash driver

### PR DESCRIPTION
**Pull request by JackChenHR**
gigadevice risc-v mcu

### CODE REVIEW

**Review by fanghuaqi — APPROVED** [review]
LGTM

**Comment by Alan-19950616** [comment]
@JackChenHR 

Hi, Jack thanks for your work on OpenOCD.

Here are some coding formatting some issues, please refer to the attachment for fixes.

[gd32vw55x_check.txt](https://github.com/riscv-mcu/riscv-openocd/files/14005554/gd32vw55x_check.txt)

File: tcl/target/gd32vw55x.cfg [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# Work-area is a space in RAM used for flash programming
+if { [info exists WORKAREASIZE] } {
+   set _WORKAREASIZE $WORKAREASIZE
+} else {
+   set _WORKAREASIZE 0x5000
+}
+
+# Allow overriding the Flash bank size
+if { [info exists FLASH_SIZE] } {
```
Comment (Alan-19950616) [inline tcl/target/gd32vw55x.cfg:21]: Don't see any use for `_FLASH_SIZE`, line 32 has flash_size set to 0x400000.

File: tcl/target/gd32vw55x.cfg [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+set _CHIPNAME riscv
+jtag newtap $_CHIPNAME cpu -irlen 5 -expected-id 0x10307a6d
+
+set _TARGETNAME $_CHIPNAME.cpu
+target create $_TARGETNAME riscv -chain-position $_TARGETNAME
+$_TARGETNAME configure -work-area-phys 0x20000000 -work-area-size 20480 -work-area-backup 0
+
+
+# Work-area is a space in RAM used for flash programming
+if { [info exists WORKAREASIZE] } {
```
Comment (Alan-19950616) [inline tcl/target/gd32vw55x.cfg:14]: Don't see any use for `_WORKAREASIZE `, line 11 has work-area-size set to 20480.

**Comment by fanghuaqi** [comment]
Hi @JackChenHR , could you do some update to your code reviewed by @Alan-19950616

**Comment by fanghuaqi** [comment]
> @JackChenHR
> 
> Hi, Jack thanks for your work on OpenOCD.
> 
> Here are some coding formatting some issues, please refer to the attachment for fixes.
> 
> [gd32vw55x_check.txt](https://github.com/riscv-mcu/riscv-openocd/files/14005554/gd32vw55x_check.txt)

Hi @Alan-19950616 you can provide some steps regarding how to check styles locally to @JackChenHR

**Comment by JackChenHR** [comment]
@Alan-19950616 Yes, please help provide a local method to check the code format. thanks!

**Comment by fanghuaqi** [comment]
Hi @JackChenHR , I checked https://openocd.org/doc-release/doxygen/patchguide.html#stepbystep this page, you can take a try.

image


@Alan-19950616 Please provide a step by step example later to check the patch style, not just the link I provided.

**Comment by fanghuaqi** [comment]
Hi @JackChenHR, here is the script you can check your commit patch

You need to install `patchutils perl`


~~~shell
# since you have commit two commits, this diff is  HEAD~2 , if you have more commits, such as 4, change to HEAD~4
git diff -U20 HEAD~2 | filterdiff -x "a/src/jtag/drivers/libjaylink/*" -x "a/tools/git2cl/*" -x "a/.github/*" | ./tools/scripts/checkpatch.pl --no-signoff -
# then you will be able to see style check output in command window
~~~

image

**Comment by fanghuaqi** [comment]
Hi @JackChenHR , I have merged the changes into our branch, thanks for your contribution, could you take a try with the nuclei/2023.10 branch, and see whether it works as expected to debug your device.

**Pull request merged**